src/workers/notifications/userGiftedPlusNotification.ts

```diff
@@ -0,0 +1,70 @@
+import { NotificationType } from '../../notifications/common';
+import { generateTypedNotificationWorker } from './worker';
+import { NotificationGiftPlusContext } from '../../notifications';
+import { SquadSource, User, UserSubscriptionFlags } from '../../entity';
+import { isGiftedPlus, isPlusMember } from '../../paddle';
+import { queryReadReplica } from '../../common/queryReadReplica';
+import { PLUS_MEMBER_SQUAD_ID } from '../userUpdatedPlusSubscriptionSquad';
+import { CioTransactionalMessageTemplateId, sendEmail } from '../../common';
+
+async function sendNotificationEmailToGifter(ctx: NotificationGiftPlusContext) {
+  const message_data = {
+    recipient_name: ctx.recipient.name,
+    recipient_image: ctx.recipient.image,
+  };
+
+  await sendEmail({
+    send_to_unsubscribed: true,
+    identifiers: { id: ctx.gifter.id },
+    transactional_message_id:
+      CioTransactionalMessageTemplateId.UserSentPlusGift,
+    to: ctx.gifter.email,
+    message_data,
+  });
+}
+
+const worker = generateTypedNotificationWorker<'user-updated'>({
+  subscription: 'api.user-gifted-plus-notification',
+  handler: async ({ user, newProfile: recipient }, con) => {
+    const { id: userId } = user;
+
+    const beforeSubscriptionFlags: Partial<UserSubscriptionFlags> = JSON.parse(
+      (user.subscriptionFlags as string) || '{}',
+    );
+    const afterSubscriptionFlags: Partial<UserSubscriptionFlags> = JSON.parse(
+      (recipient.subscriptionFlags as string) || '{}',
+    );
+
+    if (
+      isPlusMember(beforeSubscriptionFlags?.cycle) ||
+      !isGiftedPlus(afterSubscriptionFlags) ||
+      !afterSubscriptionFlags?.gifterId
```

Comment:
Should we log warning for the edge case where user is somehow plus member but he was gifted, it should not happen but still feels wrong to just ignore user's gift.

Reply:
Yes, adding a `logger.warn`

Reply:
Added ✔️